src/esp/bindings/ConfigBindings.cpp

```diff
@@ -42,6 +42,68 @@ py::object getObjectForConfigValue(const ConfigValue& value) {
   return py::cast(nullptr);
 }
 
+template <class T>
+void declareSetter(
+    pybind11::class_<esp::core::config::Configuration,
+                     esp::core::config::Configuration::ptr>& pyAbsAttr,
+    const std::string& typeName) {
+  pyAbsAttr.def(
+      "set",
+      [](Configuration& self, const std::string& key, const T val) {
+        self.set(key, val);
+      },
+      ("Set the value specified by given string key to be specified " +
```

Comment:
I'm not sure if this actually works, as far as I remember the reason `def()` accepted a plain `const char*` was because it just referenced it without actually copying anywhere. So when retrieving documentation of those, you'll get a crash due to a dangling pointer.

I might be wrong, but I personally hope it's _not_ doing a string copy for every docstring. That'd be nasty.

Reply:
@mosra  The `__init__.pyi ` has an appropriate doc string listed : 

```
    @typing.overload
    def set(self, key: str, value: str) -> None:
        """
        Set the value specified by given string key to be specified string value
        """

```
Would the problem be somewhere else?

`__init__.pyi `  also has duplicate entries for the const char* and const str& versions. I going to get rid of the const char* specializations.

Reply:
Oh, alright, then it works. I wasn't sure.